### PR TITLE
PAYARA-2336 Daily log rotation of server.log delay

### PR DESCRIPTION
**Pull request by MeroRai**

### CODE REVIEW

**Comment by MeroRai** [comment]
jenkins test please

**Comment by MeroRai** [comment]
jenkins test please

File: nucleus/core/logging/src/main/java/com/sun/enterprise/server/logging/GFFileHandler.java [hunk 1]
```diff
@@ -190,12 +191,12 @@
     String recordDateFormat;
     
     protected String logFileProperty = "";
+    private LogManager manager = LogManager.getLogManager();
+    private String cname = getClass().getName();
+    private String GFFileHander = "com.sun.enterprise.server.logging.GFFileHandler";
```
Comment (Pandrex247) [inline nucleus/core/logging/src/main/java/com/sun/enterprise/server/logging/GFFileHandler.java]: Can these be final?
They look like something that should be.

File: nucleus/core/logging/src/main/java/fish/payara/enterprise/server/logging/PayaraNotificationLogRotationTimer.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        rotationTimer.cancel();
+    }
+
+    public void restartTimer() {
+        // We will restart the timer only if the timerTask is set which
+        // means user has set a value for LogRotation based on Time
+        if (rotationTimerTask != null) {
+            rotationTimerTask.cancel();
+            rotationTimerTask = new LogRotationTimerTask(
+                    // This is wierd, We need to have a fresh TimerTask object
```
Comment (Pandrex247) [inline nucleus/core/logging/src/main/java/fish/payara/enterprise/server/logging/PayaraNotificationLogRotationTimer.java]: It's not weird :P

**Comment by MeroRai** [comment]
jenkins test please

**Pull request merged**